drivers/dma/dma_intel_adsp_hda.c

```diff
@@ -264,7 +264,9 @@ int intel_adsp_hda_dma_init(const struct device *dev)
 	data->ctx.magic = DMA_MAGIC;
 
 	pm_device_init_suspended(dev);
-	return pm_device_runtime_enable(dev);
+	(void)pm_device_runtime_enable(dev);
+
+	return 0;
```

Comment:
**_pm_device_runtime_enable_** can return other error codes when power management is enabled. Is it a good idea to ignore all return values?

Reply:
> Is it a good idea to ignore all return values?

no it is not. however, with PM disabled, this will always return -ENOSYS, which is wrong, so either the noop return value of  pm_device_runtime_enable should be changed or some logic has to be applied here before returning. Consider this a hotfix for an immediate problem for now.

Reply:
```
diff --git a/include/zephyr/pm/device_runtime.h b/include/zephyr/pm/device_runtime.h
index d14aa2fee6..99ee042891 100644
--- a/include/zephyr/pm/device_runtime.h
+++ b/include/zephyr/pm/device_runtime.h
@@ -148,13 +148,13 @@ bool pm_device_runtime_is_enabled(const struct device *dev);
 static inline int pm_device_runtime_enable(const struct device *dev)
 {
        ARG_UNUSED(dev);
-       return -ENOSYS;
+       return 0;
 }

 static inline int pm_device_runtime_disable(const struct device *dev)
 {
        ARG_UNUSED(dev);
-       return -ENOSYS;
+       return 0;
 }
```

Reply:
I would be in favor of this approach.

Reply:
done